src/languages/vala.js

```diff
@@ -21,8 +18,11 @@ function(hljs) {
         'public private protected internal ' +
         // Other
         'using new this get set const stdout stdin stderr var',
+      type:
+        'char uchar unichar int uint long ulong short ushort int8 int16 int32 int64 uint8 ' +
+        'uint16 uint32 uint64 float double bool struct enum string void',
```

Comment:
'Type' is actually intended for user-defined types that are syntactically distinguishable from other identifiers in languages where they are important structural component (like Haskell). In the case of built-in language types we deliberately just call them 'keyword'.

We should definitely improve the description of the class name, as it often causes this confusion…